ts/features/bonus/cgn/screens/merchants/CgnMerchantsListByCategory.tsx

```diff
@@ -0,0 +1,161 @@
+
+  const categoryFilter = currentCategory
+    ? {
+        productCategories: [currentCategory]
+      }
+    : {};
+
+  const initLoadingLists = () => {
+    dispatch(cgnOfflineMerchants.request(categoryFilter));
+    dispatch(cgnOnlineMerchants.request(categoryFilter));
+  };
+
+  React.useEffect(initLoadingLists, [currentCategory]);
+
+  // Mixes online and offline merchants to render on the same list
+  // merchants are sorted by name
+  const merchantsAll = useMemo(() => {
+    const onlineMerchantsValue = getValueOrElse(onlineMerchants, []);
+    const offlineMerchantsValue = getValueOrElse(offlineMerchants, []);
+
+    const merchantsAll = [...onlineMerchantsValue, ...offlineMerchantsValue];
+
+    // Removes possible duplicated merchant:
+    // a merchant can be both online and offline, or may have multiple result by offlineMerchant search API
+    const uniquesMerchants = [
+      ...new Map<OfflineMerchant["id"] | OnlineMerchant["id"], MerchantsAll>(
+        merchantsAll.map(m => [m.id, m])
+      ).values()
+    ];
+
+    return [...uniquesMerchants].sort((m1: MerchantsAll, m2: MerchantsAll) =>
+      m1.name.localeCompare(m2.name)
+    );
+  }, [onlineMerchants, offlineMerchants]);
+
+  const onItemPress = (id: Merchant["id"]) => {
+    navigation.navigate(CGN_ROUTES.DETAILS.MERCHANTS.DETAIL, {
+      merchantID: id
+    });
+  };
+
+  return (
+    <BaseScreenComponent
+      goBack
+      headerTitle={I18n.t(
+        fromNullable(categorySpecs).fold(
+          "bonus.cgn.merchantsList.navigationTitle",
+          cs => cs.nameKey
+        )
+      )}
+      contextualHelp={emptyContextualHelp}
+    >
+      {categorySpecs && (
+        <LinearGradient
+          useAngle={true}
+          angle={57.23}
+          colors={categorySpecs.colors}
+          style={[
+            IOStyles.horizontalContentPadding,
+            {
+              paddingTop: 16,
+              paddingBottom: 32
+            }
+          ]}
+        >
+          <View style={[IOStyles.row, { alignItems: "center" }]}>
+            <H1 color={"white"} style={[IOStyles.flex, { paddingRight: 30 }]}>
+              {I18n.t(categorySpecs.nameKey)}
+            </H1>
+            {categorySpecs.icon({
+              width: 57,
+              height: 57,
+              fill: IOColors.white,
+              style: { justifyContent: "flex-end" }
+            })}
+          </View>
+        </LinearGradient>
+      )}
+      {isError(onlineMerchants) && isError(offlineMerchants) ? (
+        <SafeAreaView style={IOStyles.flex}>
+          <GenericErrorComponent onRetry={initLoadingLists} />
+        </SafeAreaView>
+      ) : (
+        <View style={[IOStyles.flex, styles.listContainer]}>
+          <CgnMerchantsListView
+            merchantList={merchantsAll}
+            onItemPress={onItemPress}
+            onRefresh={initLoadingLists}
+            refreshing={
+              isLoading(onlineMerchants) || isLoading(offlineMerchants)
```

Comment:
<img src=https://user-images.githubusercontent.com/26501317/153605212-c7bcef8c-011a-42d6-a08a-baeef55685fd.png width=300 />

There is something strange on iOS with the vertical scroll indicator. It seems to me that it has to do with multiple data loads and their merge, but I have not investigated further. Maybe it's best to disable it on iOS until the cause is identified.